src/monitoring_service/handlers.py

```diff
@@ -57,9 +57,16 @@ def latest_committed_block(self) -> BlockNumber:
     def latest_confirmed_block(self) -> BlockNumber:
         return BlockNumber(self.get_latest_unconfirmed_block() - self.required_confirmations)
 
+    @property
+    def latest_timestamp(self) -> int:
+        return self.web3.eth.get_block("latest").timestamp  # type: ignore
```

Comment:
This does not work as intended. This makes "time updates" dependent on seeing new blocks. But this is exactly not possible anymore on roll-ups. Blocks gets created on-demand. This means it could be that a channel gets closed in one block and until the settle timeout expires no new block will be created. This means you clock is stuck in the past.

I'm not sure if the `context` is the correct place for this, but in fact it should be just the current system time. I just shimmed quickly through the code references to this property. It might be still necessary to access the last blocks timestamp somewhere.

Reply:
If you are sure about this, then we might have a bigger problem, because this is how the new version of the contracts are also implemented.

I did not find how to get the system time using Solidity, and the manuals have examples using `block.timestamp` to control current time.

Reply:
I resolved the part where this gets used except for the IOU stuff. I'll resolve the comment here as the main reason got addressed and there are more comments for IOUs. Hopefully this will become removed entiery soon.